src/_experimental_/GraphQL/QlHooks.re

```diff
@@ -405,6 +427,34 @@ let useTotalCollectedToken: Animal.t => option((Eth.t, BN.bn, BN.bn)) =
     | _ => None
     };
   };
+type patronLoyaltyTokenDetails = {
+  currentLoyaltyTokens: Eth.t,
+  lastCollected: BN.bn,
+  numberOfAnimalsOwned: BN.bn,
+};
+let usePatronLoyaltyTokenDetails:
+  Web3.ethAddress => option(patronLoyaltyTokenDetails) =
+  address => {
+    let (response, _) = useQueryPatronNew(address);
+
+    switch (response) {
+    | Data(responseData) =>
+      responseData##patronNew
+      ->Belt.Option.flatMap(patron =>
+          Some({
+            currentLoyaltyTokens: BN.new_("123"),
+            lastCollected: patron##lastUpdated,
+            numberOfAnimalsOwned:
+              BN.new_(patron##tokens->Obj.magic->Array.length->string_of_int),
+          })
+        )
+
+    // | Loading
+    // | Error(_error)
+    // | NoData => None
```

Comment:
Just an aside, we aren't handeling these other cases at the moment. So it is good to keep these comments here as a reminder to tighten up this code in the future.